File: brv/src/main/java/com/drake/brv/StickyAdapter.java
package com.drake.brv;

import android.view.View;

/**
 * Created by jay on 2017/12/4 上午10:52
 * <p>
 * Adds sticky headers capabilities to the {@link androidx.recyclerview.widget.RecyclerView.Adapter}. Should return {@code true} for all
 * positions that represent sticky headers.
 *
 * @link https://github.com/Doist/RecyclerViewExtensions/blob/master/StickyAdapter
 */
public interface StickyAdapter {

    boolean isStickyHeader(int position);

    interface ViewSetup {
        /**
         * Adjusts any necessary properties of the {@code holder} that is being used as a sticky header.
         * <p>
         * {@link #teardownStickyHeaderView(View)} will be called sometime after this method
         * and before any other calls to this method go through.
         */
        void setupStickyHeaderView(View stickyHeader);

        /**
         * Reverts any properties changed in {@link #setupStickyHeaderView(View)}.
         * <p>
         * Called after {@link #setupStickyHeaderView(View)}.
         */
        void teardownStickyHeaderView(View stickyHeader);
    }
}

File: brv/src/main/java/com/drake/brv/adapter/StickyAdapter.kt
package com.drake.brv.adapter

import com.drake.brv.listener.OnStickyChangeListener

/**
 * Created by jay on 2017/12/4 上午10:52
 *
 * 为RecyclerView添加悬停效果, 由 [androidx.recyclerview.widget.RecyclerView.Adapter] 实现该接口
 *
 * 原项目地址
 * @link https://github.com/Doist/RecyclerViewExtensions/blob/master/StickyAdapter
 */
interface StickyAdapter {

    var onStickyChangeListener: OnStickyChangeListener?

    /**
     * @return 是否需要悬停
     */
    fun isSticky(position: Int): Boolean
}